exatomic/va/va.py

```diff
@@ -0,0 +1,327 @@
+        freqdx = [i+1+tnmodes*j for j in range(0,2,1) for i in freqdx]
+        freqdx = np.concatenate(([0],freqdx))
+        freqdx = np.repeat(freqdx, nat)
+        modes = np.repeat(np.concatenate(([0],modes,modes)), nat)
+        symbols = np.tile(symbols, 2*nmodes+1)
+        znums = np.tile(znums, 2*nmodes+1)
+        # create dataframe
+        df = pd.DataFrame(full, columns=['x', 'y', 'z'])
+        df['freqdx'] = freqdx
+        df['Z'] = znums
+        df['symbols'] = symbols
+        df['modes'] = modes
+        self.displacements = df
+
+    def gen_gauss_inputs(self, path, routeg, routep, charge=0, mult=1, link0=''):
+        """
+        Method to write the displacements given in the displacements class variable to a
+        gaussian input file. This writes a gradient (confg*.inp) and property (confp*.inp)
+        files. As such, routeg and routep must be defined separately.
+
+        Args:
+            path (str): path to where the files will be written
+            routeg (str): gaussian route input for gradient calculation
+            routep (str): gaussian route input for property calculation
+            charge (int): charge of molecular system
+            mult (int): spin multiplicity of molecular system
+            link0 (str): link0 commands for gaussian
+        """
+        grouped = self.displacements.groupby('freqdx')
+        freqdx = self.displacements['freqdx'].drop_duplicates().values
+        n = len(str(max(freqdx)))
+        for fdx in freqdx:
+            grad_file = 'confg'+str(fdx).zfill(n)+'.inp'
+            prop_file = 'confo'+str(fdx).zfill(n)+'.inp'
+            with open(mkp(path, grad_file), 'w') as g:
+                xyz = grouped.get_group(fdx)[['symbols', 'x', 'y', 'z']]
+                g.write(_gauss_template.format(link0=link0, route=routeg, 
+                        title=str(fdx)+' gradient', charge=charge, mult=mult))
+                xyz['x'] *= Length['au', 'Angstrom']
+                xyz['y'] *= Length['au', 'Angstrom']
+                xyz['z'] *= Length['au', 'Angstrom']
+                xyz.to_csv(g, header=False, index=False, sep=' ', float_format=float_format,
+                            quoting=csv.QUOTE_NONE, escapechar=' ')
+                g.write('\n')
+            with open(mkp(path, prop_file), 'w') as p:
+                xyz = grouped.get_group(fdx)[['symbols', 'x', 'y', 'z']]
+                p.write(_gauss_template.format(link0=link0, route=routep,
+                        title=str(fdx)+' property', charge=charge, mult=mult))
+                xyz['x'] *= Length['au', 'Angstrom']
+                xyz['y'] *= Length['au', 'Angstrom']
+                xyz['z'] *= Length['au', 'Angstrom']
+                xyz.to_csv(p, header=False, index=False, sep=' ', float_format=float_format,
+                            quoting=csv.QUOTE_NONE, escapechar=' ')
+                p.write('\n')
+
+    def gen_slurm_inputs(self, path, sbatch, module, end_com=''):
```

Comment:
It seems that this method is generic enough to be place in a separate place. For example `exatomic.slurm.slurm`.

Reply:
We could probably do that, but I feel like it won't be used that much since there are the job arrays in slurm that would be more efficient I think.

Reply:
Okay, I'm open to keeping it here. I don't want the scope of this pr to creep too much.

Reply:
NB: Huge fan of a `exatomic.scheduler.slurm` concept.  At the very least templates and simple formatting.  The same applies for job array slurm scripts really.

I am curious if you looked at `exatomic.gaussian.inputs.functional_inputs` as an example of how to leverage the `Input.from_universe`  functionality, which is quite flexible for Gaussian.  If it was a restriction in the current functionality, it would be good to know where you ran into trouble to think about how to fix the design.

Reply:
What do you all think about putting that as an issue with the label feature request?

Reply:
I think we could do it. I guess I'm just worried about not making it too simple where a user ends up typing the entire file as an input. I'll try to think of some ideas to do this.